parser-typechecker/src/Unison/Server/Backend.hs

```diff
@@ -0,0 +1,557 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ViewPatterns #-}
+
+module Unison.Server.Backend where
```

Comment:
A lot of code from `HandleInput` was moved to this module. This is supposed to serve as the "backend" for both the HTTP server and the UCM command line tool. Probably needs to be broken up into smaller modules later.